### PR TITLE
Image featurization

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
It's just the same PR of the one on Justin's [repo](https://github.com/justinormont/machinelearning/pull/4)
- [x] Pass image path
- [x] Seperate ImageFeaturizing into four parts
- [x] CodeGen

### CODE REVIEW

File: src/Microsoft.ML.AutoML/Assembly.cs [hunk 1]
```diff
@@ -9,4 +9,4 @@
...
-[assembly: InternalsVisibleTo("Microsoft.ML.ModelBuilder, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
\ No newline at end of file
+[assembly: InternalsVisibleTo("Microsoft.ML.ModelBuilder.AutoMLService, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
```
Comment (JakeRadMSFT) [inline src/Microsoft.ML.AutoML/Assembly.cs]: @LittleLittleCloud which internal things do we still need ... the pipeline?

File: src/Microsoft.ML.AutoML/ColumnInference/ColumnInferenceApi.cs [hunk 2]
```diff
@@ -3,13 +3,16 @@
...
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using Microsoft.ML.Data;
 
 namespace Microsoft.ML.AutoML
 {
     internal static class ColumnInferenceApi
     {
+        private const int _maxUngroupedColumns = 20000;
```
Comment (JakeRadMSFT) [inline src/Microsoft.ML.AutoML/ColumnInference/ColumnInferenceApi.cs]: What is this for?

File: src/Microsoft.ML.AutoML/Assembly.cs (hunk 1 above)
Comment (LittleLittleCloud) [reply src/Microsoft.ML.AutoML/Assembly.cs]: yep
and TaskKind

File: src/Microsoft.ML.AutoML/ColumnInference/ColumnInferenceApi.cs (hunk 2 above)
Comment (LittleLittleCloud) [reply src/Microsoft.ML.AutoML/ColumnInference/ColumnInferenceApi.cs]: setting the threshold value for grouping columns? Model Builder doesn't use that.

**Pull request closed without merging**